File: static/pet/pixi/state_machine.js
export const PetStates = {
 IDLE:"idle",
 WORKING:"working",
 THINKING:"thinking",
 HAPPY:"happy",
 SLEEPING:"sleeping",
 CELEBRATING:"celebrating"
};

export class StateMachine {

 constructor(){
  this.state=PetStates.IDLE;
 }

 change(state){
  this.state=state;
 }

}

File: pet/pixi/state_machine.py
PET_STATES={
'idle':{'animation':'idle'},
'walking':{'animation':'walk'},
'working':{'animation':'work'},
'thinking':{'animation':'think'},
'celebrate':{'animation':'celebrate'},
'sleeping':{'animation':'sleep'}
}

def state_for_event(event):
    return PET_STATES.get(event, PET_STATES['idle'])
